pallets/msa/src/types.rs

```diff
@@ -175,12 +175,52 @@ impl AddProvider {
 	/// Create new `AddProvider`
 	pub fn new(
 		authorized_msa_id: MessageSourceId,
-		schema_ids: Option<Vec<SchemaId>>,
+		intent_ids: Option<Vec<IntentId>>,
 		expiration: BlockNumber,
 	) -> Self {
-		let schema_ids = schema_ids.unwrap_or_default();
+		let intent_ids = intent_ids.unwrap_or_default();
 
-		Self { authorized_msa_id, schema_ids, expiration }
+		Self { authorized_msa_id, intent_ids, expiration }
+	}
+}
+
+/// Structure that is signed for granting permissions to a Provider
+#[derive(TypeInfo, Clone, Debug, Decode, DecodeWithMemTracking, Encode, PartialEq, Eq)]
+pub struct AddProviderV2 {
```

Comment:
This is not used so far so the idea is to use it in future? We will need additional PRs to completely rollout this feature (including cleanup)

Reply:
I think I left this in accidentally. Was trying to decide between renaming the field in the existing struct (only affects EIP-712 signing), or generating a complete new payload type.

I think I'll remove this--no clients are using EIP-712 (verified there are no Ethereum control keys registered on Mainnet; so nobody could be signing EIP-712 payloads), so renaming the existing `schema_ids` => `intent_ids` seems the less disruptive approach.